src/ClusterManager/job_manager.py

```diff
@@ -41,6 +41,90 @@
             float("inf")),
         labelnames=("fn_name",))
 
+job_state_change_histogram = Histogram("job_state_change_latency_seconds",
+        "latency for job to change state(seconds)",
+        buckets=(1.0, 2.0, 4.0, 8.0, 16.0, 32.0, 64.0, 128.0, 256.0, 512.0, 1024.0,
+            float("inf")),
+        labelnames=("current_state",))
+
+class JobTimeRecord(object):
+    def __init__(self):
+        self.create_time = None
+        self.approve_time = None
+        self.submit_time = None
+
+
+class LRUDefatulDict(object):
```

Comment:
We might want to leverage third party caching package e.g. https://github.com/tkem/cachetools/tree/c530924cdec86855be6322d3e4dd979bfc9250e4